### ISSUE TITLE
`ConditionalList` breaks some stuff

### ISSUE THREAD

**Issue by pavelzw** [prefix-dev/recipe-format#12]
https://github.com/prefix-dev/recipe-format/blob/77da6dca6c19bda3295f4b12d462a598fcecec70/model.py#L32 I don't think this works as intended when generating a schema

`ComplexRecipe` doesn't work
```yml
# yaml-language-server: $schema=https://raw.githubusercontent.com/prefix-dev/recipe-format/main/schema.json
outputs:
  - package:
      name: my-package
      version: '0.0.1'
    build:
      noarch: python
      script:
        - echo hello world
    testaaaa: # this should fail but doesn't
      - python:
          imports:
            - polarify
```

`SimpleRecipe` works
```yml
# yaml-language-server: $schema=https://raw.githubusercontent.com/prefix-dev/recipe-format/main/schema.json
package:
  name: my-package
  version: '0.0.1'
build:
  noarch: python
  script:
    - echo hello world
testaaaa: # this actually fails
  - python:
      imports:
        - polarify
```

Maybe it makes sense to also test for failing examples?

**Comment by ytausch** [comment]
It seems like the reason is that Pydantic does not add `addidionalProperties: false` to `Output`:

https://github.com/prefix-dev/recipe-format/blob/59041fb68c3224ef97df83aeb43a25e8c1d0bc67/schema.json#L1799-L1978

**Comment by ytausch** [comment]
It seems like fixing this (by making the output object strict) fails some YAMLs that are marked as good. See #22 for reference.

**Comment by pavelzw** [comment]
Fixed in #22
